src/Symfony/Component/Form/Extension/Validator/ViolationMapper/ViolationMapper.php

```diff
@@ -131,44 +131,46 @@ public function mapViolation(ConstraintViolation $violation, FormInterface $form
 
         // Only add the error if the form is synchronized
         if ($this->acceptsErrors($scope)) {
-            $labelFormat = $scope->getConfig()->getOption('label_format');
-
-            if (null !== $labelFormat) {
-                $label = str_replace(
-                    [
-                        '%name%',
-                        '%id%',
-                    ],
-                    [
-                        $scope->getName(),
-                        (string) $scope->getPropertyPath(),
-                    ],
-                    $labelFormat
-                );
-            } else {
-                $label = $scope->getConfig()->getOption('label');
-            }
-
-            if (null === $label && null !== $this->formRenderer) {
-                $label = $this->formRenderer->humanize($scope->getName());
-            } elseif (null === $label) {
-                $label = $scope->getName();
-            }
-
-            if (false !== $label && null !== $this->translator) {
-                $label = $this->translator->trans(
-                    $label,
-                    $scope->getConfig()->getOption('label_translation_parameters', []),
-                    $scope->getConfig()->getOption('translation_domain')
-                );
-            }
-
             $message = $violation->getMessage();
             $messageTemplate = $violation->getMessageTemplate();
 
-            if (false !== $label) {
-                $message = str_replace('{{ label }}', $label, $message);
-                $messageTemplate = str_replace('{{ label }}', $label, $messageTemplate);
+            if (false !== strpos($message, '{{ label }}') || false !== strpos($messageTemplate, '{{ label }}')) {
+                $labelFormat = $scope->getConfig()->getOption('label_format');
+
+                if (null !== $labelFormat) {
+                    $label = str_replace(
+                        [
+                            '%name%',
+                            '%id%',
+                        ],
+                        [
+                            $scope->getName(),
+                            (string) $scope->getPropertyPath(),
+                        ],
+                        $labelFormat
+                    );
+                } else {
+                    $label = $scope->getConfig()->getOption('label');
+                }
+
+                if (false !== $label) {
```

Comment:
Why is this necessary? If `$label` is `null`, it will never be `false`. So the previous code was already working as expected, wasn't it?

Reply:
Yes, both versions are giving the same result. But moving up the `false !== $label` test, instead of leaving it around line 162, avoid an extra an extra test (`null === $label`).